@utahdts/utah-design-system/react/components/containers/drawer/Drawer.jsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useRef } from 'react';
+import { useImmer } from 'use-immer';
+import { joinClassNames } from '../../../util/joinClassNames';
+import { ICON_BUTTON_APPEARANCE } from '../../../enums/buttonEnums';
+import { IconButton } from '../../buttons/IconButton';
+import { useAriaMessaging } from '../../../contexts/UtahDesignSystemContext/hooks/useAriaMessaging';
+import { getFocusableElements } from '../../../util/getFocusableElements';
+import { useHandleEscape } from '../../../hooks/useHandleEscape';
+import { useHandleTab } from '../../../hooks/useHandleTab';
+import { DRAWER_PLACEMENT } from '../../../enums/drawerPlacement';
+
+/** @typedef {import('@utahdts/utah-design-system').DrawerPlacement} DrawerPlacement */
+/**
+ * @param {object} props
+ * @param {string} props.ariaLabelledBy
+ * @param {import('react').ReactNode} props.children
+ * @param {string} [props.className]
+ * @param {string} props.id
+ * @param {import('react').Ref<HTMLDivElement>} [props.innerRef]
+ * @param {import('react').KeyboardEventHandler} [props.onEscape]
+ * @param {import('react').MouseEventHandler} [props.onClose]
+ * @param {DrawerPlacement} [props.position]
+ * @returns {JSX.Element}
+ */
+export function Drawer({
+  ariaLabelledBy,
+  children,
+  className,
+  id,
+  innerRef,
+  onEscape,
+  onClose,
+  position = DRAWER_PLACEMENT.RIGHT,
+}) {
+  const ref = /** @type {typeof useRef<HTMLDialogElement | null>} */ (useRef)(null);
+  const [lastActiveElement] = useImmer(/** @type {HTMLElement | undefined} */(document.activeElement));
+  const [firstTabElement, setFirstTabElement] = useImmer(/** @type {HTMLElement | undefined} */(undefined));
+  const [lastTabElement, setLastTabElement] = useImmer(/** @type {HTMLElement | undefined} */(undefined));
+  const { addAssertiveMessage } = useAriaMessaging();
+
+  const handleEscape = useHandleEscape(onEscape);
+  const handleTab = useHandleTab(firstTabElement, lastTabElement);
+
+  useEffect(() => {
+    if (ref) {
```

Comment:
`ref` will always be "truthy". `ref.current` may be `null`. changing the `if` to be `if (ref.current)` would probably also remove the need for the `@ts-ignore`